evalml/model_understanding/graphs.py

```diff
@@ -569,6 +572,90 @@ def graph_prediction_vs_actual(y_true, y_pred, outlier_threshold=None):
+
+    children_left = est.tree_.children_left
+    children_right = est.tree_.children_right
+    features = est.tree_.feature
+    thresholds = est.tree_.threshold
+    values = est.tree_.value
+
+    def recurse(i):
+        if children_left[i] == children_right[i]:
+            return {'Name': f'Leaf_{i}'}
+        return OrderedDict({
+            'Feature': features[i],
+            'Threshold': thresholds[i],
+            'Value': values[i],
+            'Left_Child': recurse(children_left[i]),
+            'Right_Child': recurse(children_right[i])
+        })
+
+    return recurse(0)
+
+
+def visualize_decision_tree(clf, max_depth=None, rotate=False, filled=False, filepath=None):
+    """Generate an image visualizing the decision tree
+
+    Arguments:
+        clf (ComponentBase): A fitted tree-based estimator.
+        max_depth (int, optional): The depth to which the tree should be displayed.
+        rotate (bool, optional): Orient tree left to right rather than top-down.
+        filled (bool, optional): Paint nodes to indicate majority class for classification, extremity of values for
+        regression, or purity of node for multi-output.
+        filepath (str, optional): Path to where the graph should be saved. If set to None (as by default), the graph
+        will not be saved.
+
+    Returns:
+        graphviz.Source: DOT object that can be directly displayed in Jupyter notebooks.
+    """
+    if not clf.model_family == ModelFamily.DECISION_TREE:
+        raise ValueError("Tree visualizations are not supported for non-Tree estimators")
+    if max_depth and (not isinstance(max_depth, int) or not max_depth >= 0):
+        raise ValueError("Unknown value: '{}'. The parameter max_depth has to be a non-negative integer"
+                         .format(max_depth))
+    est = clf._component_obj
+
+    graphviz = import_or_raise('graphviz', error_msg='Please install graphviz to visualize trees.')
+
+    graph_format = None
+    if filepath:
+        # Cast to str in case a Path object was passed in
+        filepath = str(filepath)
+        try:
+            f = open(filepath, 'w')
+            f.close()
+        except (IOError, FileNotFoundError):
+            raise ValueError(('Specified filepath is not writeable: {}'.format(filepath)))
+        path_and_name, graph_format = os.path.splitext(filepath)
+        if graph_format:
+            graph_format = graph_format[1:].lower()  # ignore the dot
+            supported_filetypes = graphviz.backend.FORMATS
+            if graph_format not in supported_filetypes:
+                raise ValueError(("Unknown format '{}'. Make sure your format is one of the " +
+                                  "following: {}").format(graph_format, supported_filetypes))
+        else:
+            graph_format = 'pdf'  # If the filepath has no extension default to pdf
+
+    dot_data = export_graphviz(decision_tree=est, max_depth=max_depth, rotate=rotate, filled=filled)
```

Comment:
@ParthivNaresh is there a reason we don't use the data impl you wrote above this method? The pattern we've been following with other model understanding tools is to define a method which generates graph data, then define a graph method which uses the first method to get graph data and then displays that.

Reply:
@dsherry So what I understood was that the first part of this would deal just with returning a visualization for the decision tree which sklearn has already taken care of via Graphviz, and the second part would build out an intermediate data function so that an end user could call that directly for their own visualization needs (e.g. Plotly).

I could go ahead and call the intermediate data function and get the OrderedDict output and turn that into a Graphviz/text input for graphing via sklearn but that seems like a roundabout way of getting the same result, what do you think?

Reply:
Ah, got it, yeah that's fine with me. Fundamentally, there needs to be some way for a caller up the stack to generate the same visualization using javascript/other with the same data which was used for our plotly/graphviz version. As long as we've got that, and the plot method lines up with the data method in terms of behavior, I think we're good :)